components/mediation-config-admin/org.wso2.carbon.mediation.configadmin/src/main/java/org/wso2/carbon/mediation/configadmin/util/XMLPrettyPrinter.java

```diff
@@ -0,0 +1,103 @@
+/*
+* Copyright (c) 2015, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+*
+* WSO2 Inc. licenses this file to you under the Apache License,
+* Version 2.0 (the "License"); you may not use this file except
+* in compliance with the License.
+* You may obtain a copy of the License at
+*
+* http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing,
+* software distributed under the License is distributed on an
+* "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+* KIND, either express or implied. See the License for the
+* specific language governing permissions and limitations
+* under the License.
+*/
+
+package org.wso2.carbon.mediation.configadmin.util;
+
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+import org.w3c.dom.Document;
+import org.w3c.dom.ls.DOMImplementationLS;
+import org.w3c.dom.ls.LSOutput;
+import org.w3c.dom.ls.LSSerializer;
+import org.xml.sax.SAXException;
+
+import javax.xml.parsers.DocumentBuilderFactory;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.StringWriter;
+import java.io.Writer;
+
+/*
+ * This is a temporary class added to support XML Pretty Print changes for XML Comment Nodes.
+ * Original changes should go to org.wso2.carbon.utils.XMLPrettyPrinter class (in carbon kernel utils).
+ * This class should be removed after carbon kernel release 4.0.1+ with required changes.
+*/
+public class XMLPrettyPrinter {
+
+    private InputStream in;
+    private boolean xmlFormat;
+    private boolean numericEnc;
+    private boolean done = false;
+    private String encoding = "UTF-8";
+
+    private static Log log = LogFactory.getLog(XMLPrettyPrinter.class);
+
+    public XMLPrettyPrinter(InputStream in, boolean format, boolean numeric, String encoding) {
+        this.in = in;
+        xmlFormat = format;
```

Comment:
Be consistent on how you assign instance variable values here, if you used "this.in = in", then it should be "this.format = format" ..